optimade/validator/utils.py

```diff
@@ -214,29 +223,39 @@ def get(self, request: str):
 
         status_code = None
         retries = 0
+        errors = []
         # probably a smarter way to do this with requests, but their documentation 404's...
         while retries < self.max_retries:
             retries += 1
             try:
-                self.response = requests.get(self.last_request, headers=self.headers)
-            except requests.exceptions.ConnectionError as exc:
-                sys.exit(
-                    f"{exc.__class__.__name__}: No response from server at {self.last_request}, please check the URL."
+                self.response = requests.get(
+                    self.last_request,
+                    headers=self.headers,
+                    timeout=(self.timeout, DEFAULT_READ_TIMEOUT),
                 )
+
+                status_code = self.response.status_code
+                if status_code != 429:
```

Comment:
Why this specific status code? Could you add a comment or something?

Reply:
Maybe I am too familiar with this error to think it needed explaining ;)

```suggestion
                # If we have hit a 429 Too Many Requests, then try again in 1 second
                if status_code != 429:
```